awswrapper/s3.go

```diff
@@ -313,50 +321,52 @@ func (o *S3Service) UploadToS3Concurrently(content []byte, bucketName string, pa
 			}
 		}
 	}
-	_, err := uploader.Upload(&s3manager.UploadInput{
-		Bucket:      aws.String(bucketName),
-		Key:         aws.String(path),
-		Body:        bodyBytes,
-		ContentType: aws.String(http.DetectContentType(content)),
-		ACL:         aws.String(acl),
-	})
-	fmt.Println("uploading object: ", path)
+
+	uploadInput := &s3manager.UploadInput{
+		Bucket:      &bucket,
+		Key:         &path,
+		Body:        reader,
+		ContentType: contentType,
+		ACL:         &acl,
+	}
+
+	uploadOutput, err := uploader.Upload(uploadInput)
 	if err != nil {
 		if multierr, ok := err.(s3manager.MultiUploadFailure); ok {
 			// Process error and its associated uploadID
-			fmt.Println("error:", multierr.Code(), multierr.Message(), multierr.UploadID())
+			log.Fatalf("error: %s %s %s", multierr.Code(), multierr.Message(), multierr.UploadID())
 		} else {
 			// Process error generically
-			fmt.Println("error:", err.Error())
+			log.Fatalf("error: %s", err.Error())
 		}
+		return
 	}
 
-	return err
-}
+	log.Printf("Uploaded to: %s", uploadOutput.Location)
 
-// ReadFromS3Concurrently reads content from S3 concurrently
-func (o *S3Service) ReadFromS3Concurrently(bucketName string, path string) (content []byte, err error) {
-
-	session := session.Must(session.NewSession(&aws.Config{
-		Region: aws.String(o.region),
-	}))
-
-	downloader := s3manager.NewDownloader(session)
+	return
+}
 
-	var buffer aws.WriteAtBuffer
+func (o *S3Service) ReadFromS3Concurrently(bucket, path string, output io.WriterAt, options *S3Options) (err error) {
```

Comment:
Change to output io.WriterAt can be done internally instead of as an argument, right?

Reply:
Idea of accepting a `io.WriterAt` interface does have 2 purposes.

1. Calling code can maintain the lifecycle of the output stream and errors occurring when creating the stream
2. With `io.WriterAt`, we can support various number of outputs such as file, byte stream, web stream etc so that we are not limited to a file.

Reply:
Got it. I think this is a good change.